tasks/core/config.yml

```diff
@@ -37,18 +43,13 @@
       register: result
       changed_when: result is success
       notify: nextcloud update htaccess
+      failed_when: (not (result.stdout | search('successfully imported'))
+        or result.rc != 0)
```

Comment:
Instead of checking the `rc` value, you can use the `failed` test:

`result is failed`

Btw, is there a case where `occ` will fail to import but still return with an rc different to 0?

If not, let's make the `or` condition an `and` by feeding the two conditions as a list to `failed_when`.

Reply:
Yes, in my observation `occ` always reports `rc = 0` despite failing the import, here an example:

```
fatal: [hostname]: FAILED! => {
    "changed": true,
    "cmd": [
        "./occ",
        "config:import",
        "/tmp/ansible.ObLFmLnextcloud"
    ],
    "delta": "0:00:00.427724",
    "end": "2019-11-04 12:54:20.396223",
    "failed_when_result": true,
    "invocation": {
        "module_args": {
            "_raw_params": "./occ config:import \"/tmp/ansible.ObLFmLnextcloud\"",
            "_uses_shell": false,
            "argv": null,
            "chdir": "/var/www/nextcloud",
            "creates": null,
            "executable": null,
            "removes": null,
            "stdin": null,
            "warn": true
        }
    },
    "rc": 0,
    "start": "2019-11-04 12:54:19.968499",
    "stderr": "",
    "stderr_lines": [],
    "stdout": "Found invalid entries in root: logtimezone",
    "stdout_lines": [
        "Found invalid entries in root: logtimezone"
    ]
}
```

Reply:
Yes, you are right. I think we should report this as a bug to nextcloud/server. For our part, then, let's leave it as you have it.